File: TOMIMSV4/Entity/Transacciones/Ajuste/clsBe_vw_ajustes.vb
Public Class clsBe_vw_ajustes
    Implements ICloneable
    Public Property IdAjusteEnc() As Integer = 0
    Public Property IdAjusteDet As Integer
    Public Property Fecha() As String = Nothing
    Public Property Referencia() As String = ""
    Public Property Codigo_Producto() As String = ""
    Public Property Nombre_Producto() As String = ""
    Public Property IdPresentacion() As Integer = 0
    Public Property UMBas() As String = ""
    Public Property IdBodegaERP() As Integer = 0
    Public Property Codigo_Bodega() As String = ""
    Public Property Nombre_Bodega() As String = ""
    Public Property Cantidad_original() As Double = 0.0
    Public Property Cantidad_nueva() As Double = 0.0
    Public Property Peso_nuevo() As Double = 0.0
    Public Property Peso_original() As Double = 0.0
    Public Property Fecha_vence_nueva() As Date = Date.Now
    Public Property Fecha_vence_original() As Date = Date.Now
    Public Property Lote_Original() As String = ""
    Public Property Lote_Nuevo() As String = ""
    Public Property Tipo_Ajuste() As String = ""
    Public Property Modifica_Cantidad() As Boolean = False
    Public Property Enviado() As Boolean = False
    Public Property Motivo_Ajuste() As String = ""
    Public Property Observacion As String = ""
    Public Property Seccion As String = ""
    Public Property IdProductoFamilia As Integer = 0
    Public Property Nombre_Presentacion As String = ""
    Public Property Factor As Double = 0
    Public Property Codigo_Centro_Costo As String = ""
    Public Property Nombre_Centro_Costo As String = ""
    Public Property Talla As String = ""
    Public Property Color As String = ""
    Public Property Usr_Agr As String = ""

    Sub New()
    End Sub
    Sub New(ByRef IdAjusteEnc As Integer, ByVal Fecha As String, ByVal Referencia As String, ByVal Codigo_Producto As String, ByVal Nombre_Producto As String, ByVal IdPresentacion As Integer, ByVal UMBas As String, ByVal IdBodegaERP As Integer, ByVal Codigo_Bodega As String, ByVal Nombre_Bodega As String, ByVal cantidad_original As Double, ByVal cantidad_nueva As Double, ByVal peso_nuevo As Double, ByVal peso_original As Double, ByVal fecha_vence_nueva As Date, ByVal fecha_vence_original As Date, ByVal Lote_Original As String, ByVal Lote_Nuevo As String, ByVal Tipo_Ajuste As String, ByVal Modifica_Cantidad As Boolean, ByVal Enviado As Boolean, ByVal Motivo_Ajuste As String)
        Me.IdAjusteEnc = IdAjusteEnc
        Me.Fecha = Fecha
        Me.Referencia = Referencia
        Me.Codigo_Producto = Codigo_Producto
        Me.Nombre_Producto = Nombre_Producto
        Me.IdPresentacion = IdPresentacion
        Me.UMBas = UMBas
        Me.IdBodegaERP = IdBodegaERP
        Me.Codigo_Bodega = Codigo_Bodega
        Me.Nombre_Bodega = Nombre_Bodega
        Me.Cantidad_original = Cantidad_original
        Me.Cantidad_nueva = Cantidad_nueva
        Me.Peso_nuevo = Peso_nuevo
        Me.Peso_original = Peso_original
        Me.Fecha_vence_nueva = Fecha_vence_nueva
        Me.Fecha_vence_original = Fecha_vence_original
        Me.Lote_Original = Lote_Original
        Me.Lote_Nuevo = Lote_Nuevo
        Me.Tipo_Ajuste = Tipo_Ajuste
        Me.Modifica_Cantidad = Modifica_Cantidad
        Me.Enviado = Enviado
        Me.Motivo_Ajuste = Motivo_Ajuste
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function
End Class

File: WMS.EntityCore/Dtos/Ajustes/clsBe_vw_ajustes.cs
using System;

namespace WMS.EntityCore.Dtos.Ajustes
{   
    public class clsBe_vw_ajustes : ICloneable
    {
        public int IdAjusteEnc { get; set; } = 0;
        public int IdAjusteDet { get; set; }
        public string Fecha { get; set; } = string.Empty;
        public string Referencia { get; set; } = string.Empty;
        public string Codigo_Producto { get; set; } = string.Empty;
        public string Nombre_Producto { get; set; } = string.Empty;
        public int IdPresentacion { get; set; } = 0;
        public string UMBas { get; set; } = string.Empty;
        public int IdBodegaERP { get; set; } = 0;
        public string Codigo_Bodega { get; set; } = string.Empty;
        public string Nombre_Bodega { get; set; } = string.Empty;
        public double Cantidad_original { get; set; } = 0.0;
        public double Cantidad_nueva { get; set; } = 0.0;
        public double Peso_nuevo { get; set; } = 0.0;
        public double Peso_original { get; set; } = 0.0;
        public DateTime Fecha_vence_nueva { get; set; } = DateTime.Now;
        public DateTime Fecha_vence_original { get; set; } = DateTime.Now;
        public string Lote_Original { get; set; } = string.Empty;
        public string Lote_Nuevo { get; set; } = string.Empty;
        public string Tipo_Ajuste { get; set; } = string.Empty;
        public bool Modifica_Cantidad { get; set; } = false;
        public bool Enviado { get; set; } = false;
        public string Motivo_Ajuste { get; set; } = string.Empty;
        public string Observacion { get; set; } = string.Empty;
        public string Seccion { get; set; } = string.Empty;
        public int IdProductoFamilia { get; set; } = 0;
        public string Nombre_Presentacion { get; set; } = string.Empty;
        public double Factor { get; set; } = 0;
        public string Codigo_Centro_Costo { get; set; } = string.Empty;
        public string Nombre_Centro_Costo { get; set; } = string.Empty;
        public string Talla { get; set; } = string.Empty;
        public string Color { get; set; } = string.Empty;
        public string User_Agr { get; set; } = string.Empty;

        /// <summary>
        /// #CKFK20251030 Agregamos estos tres campos para la integración con ERP 
        /// ya que centro de costo maneja centro de costo dirección y departamento
        /// </summary>
        public string Centro_Costo_Erp { get; set; } = string.Empty;
        public string Centro_Costo_Dir_Erp { get; set; } = string.Empty;
        public string Centro_Costo_Dep_Erp { get; set; } = string.Empty;

        /// <summary>
        /// Constructor por defecto
        /// </summary>
        public clsBe_vw_ajustes()
        {
        }

        /// <summary>
        /// Constructor con parámetros principales
        /// </summary>
        public clsBe_vw_ajustes(
            int IdAjusteEnc,
            string Fecha,
            string Referencia,
            string Codigo_Producto,
            string Nombre_Producto,
            int IdPresentacion,
            string UMBas,
            int IdBodegaERP,
            string Codigo_Bodega,
            string Nombre_Bodega,
            double cantidad_original,
            double cantidad_nueva,
            double peso_nuevo,
            double peso_original,
            DateTime fecha_vence_nueva,
            DateTime fecha_vence_original,
            string Lote_Original,
            string Lote_Nuevo,
            string Tipo_Ajuste,
            bool Modifica_Cantidad,
            bool Enviado,
            string Motivo_Ajuste)
        {
            this.IdAjusteEnc = IdAjusteEnc;
            this.IdAjusteDet = IdAjusteDet;
            this.Fecha = Fecha ?? string.Empty;
            this.Referencia = Referencia ?? string.Empty;
            this.Codigo_Producto = Codigo_Producto ?? string.Empty;
            this.Nombre_Producto = Nombre_Producto ?? string.Empty;
            this.IdPresentacion = IdPresentacion;
            this.UMBas = UMBas ?? string.Empty;
            this.IdBodegaERP = IdBodegaERP;
            this.Codigo_Bodega = Codigo_Bodega ?? string.Empty;
            this.Nombre_Bodega = Nombre_Bodega ?? string.Empty;
            this.Cantidad_original = cantidad_original;
            this.Cantidad_nueva = cantidad_nueva;
            this.Peso_nuevo = peso_nuevo;
            this.Peso_original = peso_original;
            this.Fecha_vence_nueva = fecha_vence_nueva;
            this.Fecha_vence_original = fecha_vence_original;
            this.Lote_Original = Lote_Original ?? string.Empty;
            this.Lote_Nuevo = Lote_Nuevo ?? string.Empty;
            this.Tipo_Ajuste = Tipo_Ajuste ?? string.Empty;
            this.Modifica_Cantidad = Modifica_Cantidad;
            this.Enviado = Enviado;
            this.Motivo_Ajuste = Motivo_Ajuste ?? string.Empty;
        }

        /// <summary>
        /// Crea una copia superficial del objeto
        /// </summary>
        public object Clone()
        {
            return this.MemberwiseClone();
        }

        /// <summary>
        /// Representación en string del objeto
        /// </summary>
        public override string ToString()
        {
            return $"Ajuste: {IdAjusteEnc}-{IdAjusteDet} | {Codigo_Producto} | {Cantidad_nueva} {UMBas}";
        }
    }
}